packages/expo-camera/ios/Current/CameraView.swift

```diff
@@ -219,15 +219,28 @@ public class CameraView: ExpoView, EXCameraInterface, EXAppLifecycleListener,
     device.unlockForConfiguration()
   }
 
+  private func setIsMuted() {
+    if self.mode == .video {
+      sessionQueue.async {
+        self.session.beginConfiguration()
+        self.updateSessionAudioIsMuted()
+        self.session.commitConfiguration()
+      }
+    }
+  }
+
   private func setCameraMode() {
     sessionQueue.async {
+      self.session.beginConfiguration()
```

Comment:
How does moving these up change things? Nesting configuration blocks is fine and doesn't impact performance. iOS won't commit the changes until the outermost `commitConfiguration` is called

Reply:
hmmm I'm not sure.. I guess it wasn't necessary. However, it did fix the issue of a dead frame when switching from `picture` to `video`.

The fix that calls `updateSessionAudioIsMuted` in the `setCameraMode` is still required though -- I just thought maybe we could call `commitConfiguration` once?

Reply:
Okay, I can confirm that it does affect performance. See this code:

```swift
private func setCameraMode() {
    sessionQueue.async {
      if self.mode == .video {
        if self.videoFileOutput == nil {
          self.setupMovieFileCapture()
        }
      } else {
        self.cleanupMovieFileCapture()
      }
    }
    
    updateSessionAudioIsMuted() // This is causing dead frame
  }
```

Reply:
It might be due to `sessionQueue` and not the configuration commit  🤔. Sorry, I'm no iOS so I'm new to all this 😅

Reply:
🤔 I've pulled the code and there is still a dead frame when switching but I think thats unavoidable. As long as recording doesn't start immediately it won't be captured.

Reply:
Right. There will be little dead frame but it won't be captured. Currently, I'm adding a little delay before actually recording after switching mode. Which is fine but it would be nice for it to be handled internally without any delay. I can look at it in a separate PR :)

If you use the code above to just directly calling the mute logic, the dead frame is very noticeable and is being captured. I really think it's because of the `sessionQueue` but shouldn't hurt to call configuration commit only when necessary.

Reply:
Ok, I'll just need a bit more time to go through this. When this is called is pretty sensitive and can cause unexpected crashes when lifecycle methods are triggered.

Reply:
No worries. Feel free to edit it the best way you see fit 👍
Thanks @alanjhughes